eng/tools/TypeSpecValidation/src/TypeSpecValidation.ts

```diff
@@ -0,0 +1,75 @@
+import {exec} from "child_process";
+import fs from "fs";
+import path from "path";
+import {simpleGit} from 'simple-git';
+
+async function runCmd(cmd:string, cwd:string) {
+    console.log(`run command:${cmd}`)
+    const { err, stdout, stderr } = await new Promise((res) =>
+        exec(
+        cmd,
+        { encoding: "utf8", maxBuffer: 1024 * 1024 * 64, cwd: cwd},
+        (err: unknown, stdout: unknown, stderr: unknown) =>
+            res({ err: err, stdout: stdout, stderr: stderr })
+        )
+    ) as any;
+    let resultString = stderr + stdout;
+    console.log("Stdout output:")
+    console.log(stdout)
+    if (stderr) {
+        console.log("Stderr output:")
+        console.log(stderr)
+    }
+    if (stderr || err) {
+        throw new Error(err);
+    }
+    return resultString as string;
+}
+    
+export async function main() {
+    const args = process.argv.slice(2);
+    const folder = args[0];
+    console.log("Running TypeSpecValidation on folder:", folder);
+
+    // Verify all specs are using root level pacakge.json
+    let expected_npm_prefix = process.cwd().trim()
```

Comment:
what does trimming the cwd do?

Reply:
it's for getting rid of the newline character

Reply:
yeah but can the `cwd` ever have a new line character?

Reply:
it this not a leftover of some different resolution of the cwd where stdout was parsed